### PR TITLE
Automated platform update for springboot templates

### LINKED ISSUE #1190: Create automation for updating spring boot dependencies
Following the quarkus approach: https://github.com/knative-sandbox/kn-plugin-func/blob/main/hack/update-quarkus-platform.js 

We should have a similar automation for Spring Boot dependencies. Tasks: 
- [ ] Find out how to query for latest version on http://start.spring.io
- [ ] copy quarkus script and adapt
- [ ] Make sure that the automation script is executed when the quarkus plugin is executed (@matejvasek can you provide some pointers about the lifecycle of the `update-quarkus-platform.js` file? who is in charge of running that?

### PR DESCRIPTION
**Pull request by ningyan325**
# Changes: update-springboot-platform-on-issue-1190

Made change per issue 1190, and by following its suggestion a new set of files are added/modified (ref file changes)
- :broom: Enrich current springboot setup behavior 
-
-

/kind <kind>

Fixes #1190

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: hack/update-springboot-platform.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+const {spawn} = require('node:child_process');
+
+const cePomPath = "templates/springboot/cloudevents/pom.xml"
+const httpPomPath = "templates/springboot/http/pom.xml"
+const octokit = new Octokit({auth: process.env.GITHUB_TOKEN});
+const [owner, repo] = process.env.GITHUB_REPOSITORY.split('/')
+
+const getLatestPlatform = async () => {
+    const resp = await axios.get("https://api.github.com/repos/spring-projects/spring-boot/releases/latest")
+    return (resp.data.draft === 'false') ? resp.data.name.replace(/[A-Za-z]/g, "") : null;
```
Comment (matejvasek) [inline hack/update-springboot-platform.js:14]: Shouldn't `'false'` be just `false` ?

**Comment by matejvasek** [comment]
You will need to run `make generate/zz_filesystem_generated.go` and commit result.

File: hack/update-springboot-platform.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+const main = async () => {
+    const latestPlatform = await getLatestPlatform()
+    const prTitle = `chore: update Springboot platform version to ${latestPlatform}`
+    const branchName = `update-springboot-platform-${latestPlatform}`
+    const cePlatform = await platformFromPom(cePomPath)
+    const httpPlatform = await platformFromPom(httpPomPath)
+
+    if(latestPlatform === null) {
```
Comment (matejvasek) [inline hack/update-springboot-platform.js:107]: I would move this right below the `latestPlatform` definition.

**Comment by matejvasek** [comment]
@ningyan325 you will have to sign `EasyCLA`.

**Review by lance — COMMENTED** [review]
Looks good! Thanks for the contribution. I just have one question about octokit behavior.

File: hack/update-springboot-platform.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            per_page: perPage,
+            page: page
+        })
+
+        for (const e of resp.data) {
+            if (pred(e)) {
+                return true
+            }
+        }
+        if (resp.data.length < perPage) {
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply hack/update-springboot-platform.js:36]: It returns something that has length zero.

[the remark this answers is not in the document]
Comment (matejvasek) [reply hack/update-springboot-platform.js:36]: It returns empty array `[]`.

**Review by lance — COMMENTED** [review]
/lgtm

**Pull request merged**